gef.py

```diff
@@ -10517,9 +10522,17 @@ def canary(self) -> Optional[Tuple[int, int]]:
         auxval = self.auxiliary_vector
         if not auxval:
             return None
-        canary_location = auxval["AT_RANDOM"]
+        is_canonical = True
+        try:
+            canary_location = gef.arch.canary_address()
+        except NotImplementedError:
+            # Fall back to `AT_RANDOM`, which is the original source
+            # of the canary value but not the canonical location
+            is_canonical = False
+            canary_location = auxval["AT_RANDOM"]
         canary = gef.memory.read_integer(canary_location)
-        canary &= ~0xFF
+        if not is_canonical:
```

Comment:
Probably cleaner to remove the if here and just put all this special logic in the except case.